### PR TITLE
[gatsby-plugin-typescript] load TS preset via onCreateBabelConfig

### LINKED ISSUE #7398: [v2] TypeScript broken after `gatsby@2.0.0-beta.106`
On version `gatsby@2.0.0-beta.106` everything works.

On version `gatsby@2.0.0-beta.107` and higher there is an error for each component:

```txt
ERROR in ./src/pages/404.tsx 7:18
Module parse failed: Unexpected token (7:18)
You may need an appropriate loader to handle this file type.
| import Layout from "../components/Layout";
|
> const NotFoundPage: SFC = () => (
|   <Layout>
|     <Helmet>
 @ ./.cache/sync-requires.js 15:49-128
 @ ./.cache/app.js
 @ multi ./node_modules/react-hot-loader/patch.js (webpack)-hot-middleware/client.js?path=http://localhost:8000/__webpack_hmr&reload=true&overlay=false ./.cache/app
```

__EDIT:__

I'm using `gatsby-plugin-typescript@2.0.0-beta.8`.

### PR DESCRIPTION
**Pull request by resir014**
This hopefully fixes #7398.

### CODE REVIEW

File: packages/gatsby-plugin-typescript/src/gatsby-node.js [hunk 1]
```diff
@@ -1,15 +1,19 @@
-const { tsPresetsFromJsPresets } = require(`./`)
+// const { tsPresetsFromJsPresets } = require(`./`)
 
 const resolvableExtensions = () => [`.ts`, `.tsx`]
 
+function onCreateBabelConfig({ actions }, pluginOptions) {
+  actions.setBabelPreset({
+    name: `@babel/preset-typescript`,
```
Comment (resir014) [inline packages/gatsby-plugin-typescript/src/gatsby-node.js]: To *actually* fix this we need to remove `@babel/preset-flow` from the default presets as well. I'm not sure how to do this with the changes introduced in #6801.

**Comment by resir014** [comment]
I know it's the weekend, but I'll try to find some time to fix the tests. Sorry for the delay.

**Comment by thebigredgeek** [comment]
Hey updated the test and removed the `index.js` file.  @KyleAMathews and I were chatting and decided just to remove `babel-preset-flow` as a default so that the user can plug in whatever pre-compiler they want without having to write these kinds of hacks to remove existing transformers.

**Comment by resir014** [comment]
@thebigredgeek Awesome! I'll give a test run tomorrow.

**Comment by thebigredgeek** [comment]
Soft dependency on [#7478](https://github.com/gatsbyjs/gatsby/pull/7478)

**Comment by cpboyd** [comment]
@thebigredgeek Is there a reason you removed this, because my code doesn't build without it:
```
function onCreateWebpackConfig({ actions, loaders, stage }) {
  const jsLoader = loaders.js()
  if (
    !(
      jsLoader &&
      jsLoader.loader
    )
  ) {
    return
  }
  actions.setWebpackConfig({
    module: {
      rules: [
        {
          test: /\.tsx?$/,
          use: [
            {
              loader: jsLoader.loader,
            },
          ],
        },
      ],
    },
  })
}
 exports.onCreateWebpackConfig = onCreateWebpackConfig
```

My specific bug is the same as my [comment](https://github.com/gatsbyjs/gatsby/issues/7398#issuecomment-413948133) on issue #7398

**Comment by KyleAMathews** [comment]
@cpboyd this bug still exists if you upgrade to the latest version?

**Comment by azdanov** [comment]
Still doesn't work on latest `rc`.

```js
{
	...
    "gatsby": "2.0.0-rc.0",
    "gatsby-plugin-typescript": "^2.0.0-rc.0",
	...
}
```

**Comment by valentine** [comment]
I'm having problems as well:
```
Module parse failed: The keyword 'interface' is reserved (13:0)
You may need an appropriate loader to handle this file type.
```

Putting back the `onCreateWebpackConfig` function fixed it for me.

**Comment by resir014** [comment]
Apologies eveyone - I could push a hotfix, but not until a couple more hours.

**Comment by cpboyd** [comment]
@KyleAMathews I just did a yarn upgrade to the latest RC and still have the error without adding `onCreateWebpackConfig`

**Comment by XHMM** [comment]
I upgraded **gatsby** and **gatsby-plugin-typescript** to **2.0.0-rc.0** , and has the same error #7462 @KyleAMathews

**Comment by resir014** [comment]
Really sorry for the delay guys - #7525.

**Pull request merged**